bucket_manifest/bucket_manifest_job.py

```diff
@@ -0,0 +1,216 @@
+    with Pool(NUMBER_OF_THREADS) as pool:
+        pool.map(par_submit_job, keys)
+
+
+def list_objects(bucket_name):
+    """
+    List all objects in the bucket
+    """
+    result = []
+
+    with open("./creds.json") as creds_file:
+        creds = json.load(creds_file)
+        aws_access_key_id = creds.get("aws_access_key_id")
+        aws_secret_access_key = creds.get("aws_secret_access_key")
+
+    client = boto3.client(
+        "s3",
+        region_name=REGION,
+        aws_access_key_id=aws_access_key_id,
+        aws_secret_access_key=aws_secret_access_key,
+    )
+
+    try:
+        paginator = client.get_paginator("list_objects_v2")
+        logging.info("start to list objects in {}".format(bucket_name))
+        pages = paginator.paginate(Bucket=bucket_name, RequestPayer="requester")
+        for page in pages:
+            for obj in page["Contents"]:
+                result.append(obj["Key"])
+    except ClientError as e:
+        logging.error(
+            "Can not list objects in the bucket {}. Detail {}".format(bucket_name, e)
+        )
+
+    return result
+
+
+def write_messages_to_tsv(queue_url, n_total_messages, bucket_name):
+    """
+    Consume the sqs and write results to tsv manifest
+
+    Args:
+        queue_url(str): SQS url
+        n_total_messages(int): The expected number of messages being received
+        bucket_name(str): bucket for uploading the manifest to
+    """
+    with open("./creds.json") as creds_file:
+        creds = json.load(creds_file)
+        aws_access_key_id = creds.get("aws_access_key_id")
+        aws_secret_access_key = creds.get("aws_secret_access_key")
+
+    logging.info("Start consuming queue {}".format(queue_url))
+    # Create SQS client
+    sqs = boto3.client("sqs", region_name=REGION)
+
+    n_messages = 0
+    files = []
+    while True:
+        try:
+            # recive a message from SQS queue
+            response = sqs.receive_message(
+                QueueUrl=queue_url,
+                AttributeNames=["SentTimestamp"],
+                MaxNumberOfMessages=1,
+                VisibilityTimeout=0,
+                WaitTimeSeconds=0,
+            )
+
+            n_messages += len(response["Messages"])
+            if n_messages % 10 == 0:
+                logging.info("Received {}/{} messages".format(n_messages, n_total_messages))
+
+            for message in response["Messages"]:
+                msgBody = json.loads(message["Body"])
+                if "ERROR" in msgBody:
+                    msgBody["size"] = 0
+                    msgBody["md5"] = msgBody["ERROR"]
+                    del msgBody["ERROR"]
+                files.append(json.loads(message["Body"]))
+
+                # Delete received message from queue
+                receipt_handle = message["ReceiptHandle"]
+                sqs.delete_message(QueueUrl=queue_url, ReceiptHandle=receipt_handle)
+
+            if n_messages >= n_total_messages:
```

Comment:
instead of breaking here, you could do `while n_messages >= n_total_messages:` instead of `while True:` which might be a little cleaner